src/authentication/authentication.service.ts

```diff
@@ -44,4 +54,54 @@ export class AuthenticationService {
       token: this.generateAuthToken({ email: user.email, sub: user.id }),
     }
   }
+
+  async updatePasswordWithToken(
+    email: string,
+    token: string,
+    password: string,
+  ) {
+    let user
+    try {
+      this.jwtService.verify(token)
+      user = await this.usersService.findUserByResetToken(email, token)
+    } catch (e) {
+      if (
+        e instanceof SyntaxError || // token parse failed
+        e instanceof PrismaClientKnownRequestError // token doesn't match a user
+      ) {
+        throw new BadRequestException('Invalid token')
```

Comment:
Should this be a `403 Forbidden` since the token (method of authorization) is invalid? 🤔